### PR TITLE
(GH-588) Add option to search packages in all available repositories

### PR DESCRIPTION
**Pull request by mwallner**
see #588

### CODE REVIEW

**Comment by gep13** [comment]
@mwallner am I missing something here?  Or where are you actually telling it to search all repositories?  Perhaps we can talk about this one when we have a chat?

**Comment by mwallner** [comment]
hey @gep13 - since I'm pretty busy currently here the explanation:
https://github.com/chocolatey/ChocolateyGUI/blob/1a34f8294914877b21c37e96d2af8d14a423c557/Source/ChocolateyGui/Services/ChocolateyService.cs#L168
(the logic is inverse - if "search in all repos" is active - the source to look in is _not_ set - otherwise it will be fixed to the currently active)

File: Source/ChocolateyGui/Models/AppConfiguration.cs [hunk 1]
```diff
@@ -17,5 +17,7 @@ public class AppConfiguration
         public bool DefaultToTileViewForRemoteSource { get; set; }
 
         public bool UseDelayedSearch { get; set; }
+
+        public bool SearchInAllRepositories { get; set; }
```
Comment (gep13) [inline Source/ChocolateyGui/Models/AppConfiguration.cs]: Is this the right terms here?  Chocolatey GUI doesn't refer to repositories, but rather Sources.  Should this be `SearchInAllRepositories`?

File: Source/ChocolateyGui/Models/PackageSearchOptions.cs [hunk 2]
```diff
@@ -15,7 +15,8 @@ public struct PackageSearchOptions
             bool includePrerelease,
             bool includeAllVersions,
             bool matchWord,
-            string source)
+            string source,
+            bool searchInALlRepos)
```
Comment (gep13) [inline Source/ChocolateyGui/Models/PackageSearchOptions.cs]: In addition to the other comment, this should be `searchInAllRepos`

File: Source/ChocolateyGui/Properties/Resources.resx [hunk 3]
```diff
@@ -642,4 +642,7 @@ NOTE: Probably only necessary to change in RTL languages.</comment>
   <data name="PackageView_Version" xml:space="preserve">
     <value>Version</value>
   </data>
+  <data name="SettingsView_SearchInAllRepos" xml:space="preserve">
+    <value>Search in all Repos</value>
```
Comment (gep13) [inline Source/ChocolateyGui/Properties/Resources.resx]: Depending on the outcome of the above, we will likely want to change this text as well.

**Comment by gep13** [comment]
@mwallner I definitely prefer this being an opt-in option, so based on that, I have no objections.

Can you have a look at the comments that I made, and also update to the latest develop branch?  There are some changes that you will need to apply in the SettingsView to include a new description for the setting.  Let me know if you have a chance to do this.

Apologies in the time taken to get this reviewed.

File: Source/ChocolateyGui/Models/AppConfiguration.cs (hunk 1 above)
Comment (punker76) [reply Source/ChocolateyGui/Models/AppConfiguration.cs]: I'm with @gep13 , so `SearchInAllSources` could be the name...

**Comment by DamianMaslanka5** [comment]
@gep13 @mwallner Instead of enabling this option globally I would like to see `All` under `This PC`.
Right now when i go to private source i see my packages, and if I want to search for a specific package I can't when `Search in all Repos` is enabled. I need to go to options, disable it and then search again.

Probably there should be a way to hide `All`?

**Comment by mwallner** [comment]
@DamianMaslanka5 - great idea!
@gep13 - I will see if I can pull this up .. :-)

**Comment by mwallner** [comment]
@gep13 @DamianMaslanka5  - see GH-646
I'd be happy to close this PR if you're ok with the new one

**Comment by gep13** [comment]
Closing in favour of #646

**Pull request closed without merging**